pkg/operator/controller/ingress/status.go

```diff
@@ -392,6 +393,23 @@ func computeIngressDegradedCondition(conditions []operatorv1.OperatorCondition)
 		},
 	}
 
+	// Only check the default ingress controller for the canary
+	// success status condition.
+	if icName == manifests.DefaultIngressControllerName {
```

Comment:
This doesn't strictly need to be conditionalized because the logic below skips missing status conditions.

Reply:
I think it makes sense to place this conditional here to make it obvious that the canary condition only applies to the default ingress controller.